.github/workflows/smoketest.yaml

```diff
@@ -4,9 +4,14 @@ on:
   issue_comment:
     types: [created] # Add "smoke test" as a PR comment to trigger this workflow.
 
+# Permissions needed for reacting and adding comments for IssueOps commands
+# See: https://github.com/github/branch-deploy/blob/48285b12b35e47e2dde0c27d2abb33daa846d98b/README.md?plain=1#L189-L197
 permissions:
-  contents: read
-  pull-requests: write # For adding a reaction to the comment
+  pull-requests: write # Required for commenting on PRs
+  deployments: write # Required for updating deployment statuses
+  contents: write # Required for reading/writing the lock file
```

Comment:
This makes me wonder if we really need workflow. On one hand it is carefully reviewed IssueOps, on the other hand it is a potentially untrusted code running with high privileges...

Reply:
The idea is that the branch-deploy action makes this safe to trigger with an issue comment. It's the solution that we've been recommending to other open source maintainers.

I intended this to run on the code from the PR, not the base code. That was a mistake.

Reply:
Good catch, I didn't notice it checkouts the main instead of `steps.branch-deploy.outputs.sha`. @kevinbackhouse could you remind us what is the purpose of the workflow?

Reply:
It's for testing that the examples still work. It needs the `COPILOT_TOKEN` secret so a standard pull-request trigger doesn't work. I figure we could run it on-demand by adding a comment to a PR.

Reply:
I've updated it to use `github.head_ref`.